catalyst/runners/runner.py

```diff
@@ -670,6 +670,56 @@ def on_loader_end(self, runner):
         for batch in loader:
             yield self.predict_batch(batch)
 
+    def evaluate_loader(
+        self,
+        loader: DataLoader,
+        callbacks: "Union[List[Callback], OrderedDict[str, Callback]]",
+        model: Optional[Model] = None,
+        seed: int = 42,
+        verbose: bool = False,
+    ) -> Dict:
+        """
+        Evaluates data from loader with given model and returns obtained metrics. # noqa: DAR401
+
+        Args:
+            loader: loader to predict
+            callbacks: list or dictionary with catalyst callbacks
+            model: model, compatable with current runner.
+                If `None` simply takes current model from runner.
+            seed: random seed to use before prediction
+            verbose: if `True`, it displays the status of the evaluation to the console.
+
+        Returns:
+            Dict with metrics counted on the loader.
+        """
+        if isinstance(callbacks, List):
+            for callback in callbacks:
+                if isinstance(callback, CheckpointCallback):
+                    raise RunnerException(
+                        "CheckpointCallback isn`t allowed for evaluation loader method"
+                    )
+        else:
+            for callback in callbacks.values():
+                if isinstance(callback, CheckpointCallback):
+                    raise RunnerException(
+                        "CheckpointCallback isn`t allowed for evaluation loader method"
+                    )
```

Comment:
May I ask, Why is CheckpointCallback not allowed for evaluation?

Reply:
@penguinflys, thanks for the question! 
I think the idea behind evaluation_loader functionality is to minimize behavior to "make a loader inference and return metrics, not more." In the beginning, we decided to reduce functionality because of the "good first issue" label =) 
If you have any suggestions, feel free to share them!

Reply:
I think `CheckpointCallback` could be very helpful if the user makes a prediction with the trained model.

Reply:
`resume` feature is under development
we have a few questions to solve before its an introduction
it's a bit complicated due to the fact, that user could use any possible PyTorch checkpoints ;)
for such a reason loading the model is out of scope for now

Reply:
thanks for feedbacking